team-validator.js

```diff
@@ -253,16 +253,15 @@ class Validator {
 				if (!set.moves[i]) continue;
 				let move = tools.getMove(Tools.getString(set.moves[i]));
 				if (!move.exists) return ['"' + move.name + '" is an invalid move.'];
-				set.moves[i] = move.name;
```

Comment:
`set.moves` can be either name or ID until after validation. You should do `= move.id` if you want that to be the new guarantee.